### PR TITLE
fix: perform home page redirection on server to prevent content flashing

### PR DESCRIPTION
**Pull request by matthieu-foucault**

### CODE REVIEW

**Review by pbastia — DISMISSED** [review]
Needs rebase

very nice!

File: app/pages/index.tsx [hunk 1]
```diff
@@ -7,12 +7,28 @@ import DefaultLayout from 'layouts/default-layout';
...
+  if (landingRoute === '/')
+    return {
+      props: {}
+    };
+
+  return {
+    redirect: {destination: landingRoute, permanent: false}
+  };
+}
+
```
Comment (kriscooke) [inline app/pages/index.tsx:31]: I'm pretty sure the routing needs to be done in `server/index.js` - even using getServerSideProps, this only runs when you are on the homepage.

So if I'm on an application page for example and click "Dashboard", I don't hit this until I first load the homepage on the client.

**Review by kriscooke — CHANGES_REQUESTED** [review]

File: app/pages/index.tsx (hunk 1 above)
Comment (kriscooke) [reply app/pages/index.tsx:31]: (in other words, using getServerSideProps only makes the homepage load faster, and thus perform the redirect a bit faster)

Comment (matthieu-foucault) [reply app/pages/index.tsx:31]: `getServerSideProps` is run by the next middleware on the server side, so this should work

Comment (kriscooke) [reply app/pages/index.tsx:31]: Note to self: This solution doesn't work in development when using the `AS_REPORTER` etc. flags, which won't be the case on production.

**Review by kriscooke — APPROVED** [review]

**Pull request merged**